Set up location driven Geotriggers sample (#239)
Esri/arcgis-maps-sdk-flutter-samples: lib/samples/set_up_location_driven_geotriggers/README.md

> # Set up location driven geotriggers

Reviewer: ```suggestion
# Set up location-driven geotriggers
```

Author: So, there is a problem in our README parser for the title. It does not like hyphens. This change will break the README metadata checker.

Reviewer: Oh interesting. It's funny as I was reviewing a new sample yesterday that has a hyphen in the title too, I think there's one other as well that we haven't got to. Ok - let's leave it out here then and I'll write it up separately, will see how other SDKs have allowed for this in the readme scripts.

Create a notification every time a given location data source has entered and/or exited a set of features or graphics.

> ![Geotriggers](set_up_location_driven_geotriggers.png)

Reviewer: ```suggestion
![Image of set up location-driven geotriggers](set_up_location_driven_geotriggers.png)
```

## Use case

Geotriggers can be used to notify users when they have entered or exited a geofence by monitoring a given set of features or graphics. They could be used to display contextual information to museum visitors about nearby exhibits, notify hikers when they have wandered off their desired trail, notify dispatchers when service workers arrive at a scene, or more.

## How to use the sample

Observe a virtual walking tour of the Santa Barbara Botanic Garden. Information about the user's current Garden Section, as well as information about nearby points of interest within 10 meters will display or be removed from the UI when the user enters or exits the buffer of each feature.

## How it works

1. Create a `GeotriggerFeed` with a `LocationDataSource` class (in this case, a `SimulatedLocationDataSource`).

> 2. Create a `FeatureFenceParameters` class from a `ServiceFeatureTable`, a buffer distance at which to monitor each feature, an Arcade Expression, and a name for the specific geotrigger.

Reviewer: ```suggestion
2. Create a `FeatureFenceParameters` class from a `ServiceFeatureTable`, a buffer distance at which to monitor each feature, an `ArcadeExpression`, and a name for the specific geotrigger.
```
Appreciate this is a miss in the design too but probably worth the notation. Probably not worth a design update just for this though.

3. Create a `FenceGeotrigger` with the geotrigger feed, a `FenceRuleType`, and the fence parameters.
4. Create a `GeotriggerMonitor` with the fence geotrigger and call `GeotriggerMonitor.start()` to begin listening for events that meet the `FenceRuleType`.
5. When a `GeotriggerMonitor.geotriggerNotification` emits, capture the `GeotriggerNotificationInfo`.

> 6. For more information about the feature that triggered the notification, cast the `GeotriggerNotificationInfo` to a `FenceGeotriggerNotificationInfo` and call `FenceGeotriggerNotificationInfo::fenceGeoElement()`.

Reviewer: ```suggestion
6. For more information about the feature that triggered the notification, cast the `GeotriggerNotificationInfo` to a `FenceGeotriggerNotificationInfo` and call `FenceGeotriggerNotificationInfo.fenceGeoElement`.
```

> 7. Depending on the `FenceGeotriggerNotificationInfo::fenceNotificationType()` display or hide information on the UI from the `GeoElement`'s attributes.

Reviewer: ```suggestion
7. Depending on the `FenceGeotriggerNotificationInfo.fenceNotificationType` display or hide information on the UI from the `GeoElement`'s attributes.
```
